couchbase/tests/test_integration.py

```diff
@@ -61,11 +60,9 @@ def test_query_monitoring_metrics(aggregator, dd_run_check, instance_query, couc
     couchbase = Couchbase('couchbase', {}, [instance_query])
     dd_run_check(couchbase)
 
-    query_stats_optional = set(QUERY_STATS).difference(QUERY_STATS_ALWAYS_PRESENT)
-    for mname in QUERY_STATS_ALWAYS_PRESENT:
-        aggregator.assert_metric('couchbase.query.{}'.format(mname), tags=CHECK_TAGS, count=1)
-    for mname in query_stats_optional:
-        aggregator.assert_metric('couchbase.query.{}'.format(mname), tags=CHECK_TAGS, at_least=0)
+    for metric_name in QUERY_STATS:
```

Comment:
If I'm not mistaken, QUERY_STATS is defined in the code and not in the tests. Personally, I prefer to use only data and constants defined in the test environment in tests. This way if someone accidentally changes something in QUERY_STATS (in the production code) my tests will detect it.
It may seem like code repetition, but in reality it is not. We are completely separating the tests from the code and only joining them by the assertions on the output.
What do you think? Does it make sense to you?

Reply:
Yes it does. This is what I've done for [these metrics](https://github.com/DataDog/integrations-core/pull/16344/files#diff-5d1a2f80cd3582821be9cc9972416513643a27e9fa635a5d69b6ed117db10d5dR565-R587). Let me modify this set as well!